salt/pillar/rethinkdb_pillar.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+'''
+Provide external pillar data from RethinkDB
+
+:depends: rethinkdb (on the salt-master)
+
+
+salt master rethinkdb configuration
+===================================
+These variables must be configured in your master configuration file.
+    * ``rethinkdb.host`` - The RethinkDB server. Defaults to ``'salt'``
+    * ``rethinkdb.port`` - The port the RethinkDB server listens on.
+      Defaults to ``'28015'``
+    * ``rethinkdb.database`` - The database to connect to.
+      Defaults to ``'salt'``
+    * ``rethinkdb.username`` - The username for connecting to RethinkDB.
+      Defaults to ``''``
+    * ``rethinkdb.password`` - The password for connecting to RethinkDB.
+      Defaults to ``''``
+
+
+salt-master ext_pillar configuration
+====================================
+
+The ext_pillar function arguments are given in single line dictionary notation.
+
+.. code-block:: yaml
+
+  ext_pillar:
+    - rethinkdb: {table: ext_pillar, id_field: minion_id, field: pillar_root, pillar_key: external_pillar}
+
+In the example above the following happens.
+    * The salt-master will look for external pillars in the 'ext_pillar' table
+      on the RethinkDB host
+    * The minion id will be matched against the 'minion_id' field
+    * Pillars will be retrieved from the nested field 'pillar_root'
+    * Found pillars will be merged inside a key called 'external_pillar'
+
+
+Module Documentation
+====================
+'''
+from __future__ import absolute_import
+
+# Import python libraries
+import logging
+
+# Import 3rd party libraries
+try:
+    import rethinkdb as r
+    HAS_RETHINKDB = True
+except ImportError:
+    HAS_RETHINKDB = False
+
+__virtualname__ = 'rethinkdb'
+
+__opts__ = {
+    'rethinkdb.host': 'salt',
+    'rethinkdb.port': '28015',
+    'rethinkdb.database': 'salt',
+    'rethinkdb.username': None,
+    'rethinkdb.password': None
+}
+
+
+def __virtual__():
+    if not HAS_RETHINKDB:
+        return False
+    return True
+
+
+# Configure logging
+log = logging.getLogger(__name__)
+
+
+def ext_pillar(minion_id,
+               pillar,
+               table='pillar',
+               id_field=None,
+               field=None,
+               pillar_key=None):
+    '''
+    Collect minion external pillars from a RethinkDB database
+
+Arguments:
```

Comment:
This line should be indented to match up with the rest of the doc section.

Reply:
Done